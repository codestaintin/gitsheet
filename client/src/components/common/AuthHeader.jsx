import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const AuthHeader = ({ logout }) => (
  <Fragment>
    <nav
      className="navbar navbar-light"
      style={{ backgroundColor: '#3D4E81' }}
    >
      <ul className="navbar-nav">
        <li className="nav-item">
          <Link className="nav-link link" to="/">Git Sheet</Link>
        </li>
      </ul>

      <ul className="nav justify-content-end">
        <li className="nav-item">
          <Link
            to="#"
            className="nav-link link"
          >
            <button
              className="btn btn-outline-light"
              type="button"
              onClick={logout}
            >
              Log Out
            </button>
          </Link>
        </li>
      </ul>
    </nav>
  </Fragment>
);

AuthHeader.propTypes = {
  logout: PropTypes.func.isRequired
};

export default AuthHeader;