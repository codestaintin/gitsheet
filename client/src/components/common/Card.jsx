import React, { Fragment } from 'react';

const Card = () => (
  <Fragment>
    <div className="card border-primary">
      <div className="card-header">Category</div>
      <div className="card-body">
        <p className="card-text">
        This is a longer card with supporting text
        below as a natural lead-in to additional content.
        This content is a little bit longer.
        </p>
      </div>
    </div>
  </Fragment>
);

export default Card;