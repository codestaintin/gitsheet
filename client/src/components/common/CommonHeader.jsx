import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const Header = ({ sidebarOpen }) => (
  <Fragment>
    <nav
      className="navbar navbar-light"
      style={{ backgroundColor: '#3D4E81', zIndex: sidebarOpen ? 0 : 100 }}
    >
      <ul className="navbar-nav">
        <li className="nav-item">
          <Link className="nav-link link" to="/">Git Sheet</Link>
        </li>
      </ul>

      <ul className="nav justify-content-end">
        <li className="nav-item">
          <Link className="nav-link link" to="/signin">Sign In</Link>
        </li>
        <li className="nav-item">
          <Link className="nav-link link" to="/signup">Sign Up</Link>
        </li>
      </ul>
    </nav>
  </Fragment>
);

Header.propTypes = {
  sidebarOpen: PropTypes.bool
};

export default Header;