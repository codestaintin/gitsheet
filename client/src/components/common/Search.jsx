import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const Search = ({ handleSearch }) => (
  <div className="container">
    <div className="row">
      <div className="col-12">
        <div className="input-group">
          <input
            className="form-control border-secondary py-2"
            type="search"
            placeholder="Search for a git command"
            name="searchTerm"
            onChange={handleSearch}
          />
          <div className="input-group-append">
            <button className="btn btn-outline-dark" type="button">
              <i className="fa fa-search" />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
);

Search.propTypes = {
  handleSearch: PropTypes.func
};

export default Search;