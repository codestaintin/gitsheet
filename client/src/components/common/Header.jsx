import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { logOut } from '../../actions/auth/singinAction';
import AuthHeader from './AuthHeader';
import CommonHeader from './CommonHeader';

/**
 * @export
 *
 * @class Header
 *
 * @extends {React.Component}
 */
export class Header extends Component {
  constructor(props) {
    super(props);
    this.logoutUser = this.logoutUser.bind(this);
  }

  logoutUser() {
    this.props.logOut();
  }

  render() {
    const { sidebarOpen, signIn } = this.props;
    return (
      <Fragment>
        { (signIn.success)
          ? <AuthHeader logout={this.logoutUser} sidebarOpen={sidebarOpen} />
          : <CommonHeader sidebarOpen={sidebarOpen} />
        }
      </Fragment>
    );
  }
}

Header.propTypes = {
  logOut: PropTypes.func,
  sidebarOpen: PropTypes.bool,
  signIn: PropTypes.object
};

const mapStateToProps = state => ({
  signIn: state.authReducer
});

const mapDispatchToProps = dispatch => bindActionCreators({ logOut }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Header);