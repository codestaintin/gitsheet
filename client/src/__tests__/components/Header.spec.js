import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/common/Header';

describe('<Header>', () => {
  const props = {
    logOut: jest.fn(() => {}),
    signIn: {},
    sidebarOpen: true
  };
  const tree = shallow(<Header {...props} />);

  it('should render the Header component', () => {
    expect(tree).toMatchSnapshot();
  });
  it('should simulate logoutUser', () => {
    const spy = jest.spyOn(tree.instance(), 'logoutUser');
    tree.instance().logoutUser();
    expect(spy).toHaveBeenCalled();
  });
});