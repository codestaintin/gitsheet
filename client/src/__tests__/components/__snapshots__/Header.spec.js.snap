// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> should render the Header component 1`] = `
<Fragment>
  <Header
    sidebarOpen={true}
  />
</Fragment>
`;
